AirCasting/Map/AirMapView.swift

```diff
@@ -11,44 +11,40 @@ import Foundation
 import CoreData
 
 struct AirMapView: View {
-    var thresholds: [SensorThreshold]
+    var threshold: SensorThreshold
     @ObservedObject var session: SessionEntity
-
-    private var measurementStream: MeasurementStreamEntity? {
-        if session.type == .mobile && session.deviceType == .MIC {
-            return session.dbStream
-        } else {
-            #warning("Select proper measurementStream")
-            return session.measurementStreams?.firstObject as? MeasurementStreamEntity
-        }
-    }
-
+    @Binding var selectedStream: MeasurementStreamEntity?
+    
     private var pathPoints: [PathPoint] {
-        measurementStream?.allMeasurements?.compactMap {
-            if let location = $0.location {
-                return PathPoint(location: location, measurement: $0.value)
-            } else {
-                #warning("TODO: Do something with no location points")
-                return nil
-            }
-       } ?? []
+        if let measurementStream = selectedStream {
```

Comment:
Since you're making `?? []` anyway in the next statement you can just omit this `if let` and add one more item to optional chaining like this (also note the use of `guard let` to get rid of unneeded indentation): 
``` Swift
return selectedStream?.allMeasurements?.compactMap {
     #warning("...")
     guard let location = $0.location else { return nil }
     return PathPoint(location: location, measurement: $0.value)
} ?? []
```